nav2_behavior_tree/include/nav2_behavior_tree/plugins/condition/is_error_code_active_condition.hpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2022 Joshua Wallace
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__IS_ERROR_CODE_ACTIVE_CONDITION_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__CONDITION__IS_ERROR_CODE_ACTIVE_CONDITION_HPP_
+
+#include <string>
+#include <memory>
+#include <vector>
+#include <set>
+
+#include "rclcpp/rclcpp.hpp"
+#include "behaviortree_cpp_v3/condition_node.h"
+
+namespace nav2_behavior_tree
+{
+
+class IsErrorCodeActive : public BT::ConditionNode
+{
+public:
+  IsErrorCodeActive(
+    const std::string & condition_name,
+    const BT::NodeConfiguration & conf)
+  : BT::ConditionNode(condition_name, conf)
+  {
+    getInput<std::set<int>>("error_codes_to_check", error_codes_to_check_);
+  }
+
+  IsErrorCodeActive() = delete;
+
+  BT::NodeStatus tick()
```

Comment:
Petty, but I would prefer the `tick()` to be in the cpp file since that's where I'm going to always look for the functional implementations. 

The  constructor too while we're at it

Reply:
So, 

the issue with moving it to the cpp file is that the other conditions will need to link against that file which is blocked because of registering the node call. (can't register the same node twice).  I could make a impl.hpp file if you would like?

Reply:
No that's fine :+1: